### PR TITLE
MWPW-105618 - Consonant Icon block

### PR DESCRIPTION
**Pull request by Sartxi**
* Added support for Consonant 'Icon' blocks

Resolves: [MWPW-105618](https://jira.corp.adobe.com/browse/MWPW-105618)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/sarchibeque/icon-block
- After: https://consonant-icon-block--milo--adobecom.hlx.page/drafts/sarchibeque/icon-block

### CODE REVIEW

File: libs/blocks/icon-block/icon-block.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    decorateBlockBg(el, children[0]);
+  }
+  const foreground = document.createElement('div');
+  foreground.classList.add('foreground', 'container');
+  el.appendChild(foreground);
+  return foreground;
+}
+
+function decorateContent(row, isVertical) {
+  if (row) {
```
Comment (chrischrischris) [inline libs/blocks/icon-block/icon-block.js:31]: nit: early return
```suggestion
  if (!row) return;
```

**Review by chrischrischris — APPROVED** [review]
✅  just two nits

File: libs/blocks/icon-block/icon-block.css [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+.icon-block .foreground.container .text .action-area>a:last-child {
+  margin: 0;
+}
+
+
+.icon-block .foreground.container div {
+  flex: 1 1 auto;
+}
+
+/* Vertical */
```
[the remark this answers is not in the document]
Comment (Sartxi) [reply libs/blocks/icon-block/icon-block.css]: Thanks, okay I removed the comment

**Review by auniverseaway — CHANGES_REQUESTED** [review]
I think all the two-up, three-up, four-up and related code needs to be kicked out. This should be handled in section-metadata now.

You all built these when we didn't really have a concept of sections. Now that we do, we can put those styles in there.

```css
.section-metadata.two-up {
  display: grid;
  grid-template-columns: 1fr 1fr;
}
```
...or something like that.

**Comment by amauch-adobe** [comment]
@Sartxi , were you able to make the changes requested by Chris Millar as mentioned in his comment above?  I'm more than willing to look over this PR and do QA, but I just wanted to make sure I'm looking/testing what really will go out.

**Comment by amauch-adobe** [comment]
@Sartxi , just notify @auniverseaway when you are ready for him to relook over his requested changes.

**Comment by Sartxi** [comment]
Hello @auniverseaway I have made updates so that each icon block is now in a segment and uses the style row to set the number of grid columns. @ryanmparrish and I thought it would be good to go with a grid class to apply the proper styles to the segment blocks grid container. Let me know your thoughts. Thanks!

**Review by amauch-adobe — CHANGES_REQUESTED** [review]
@Sartxi , it looks like there is a responsive design issue with viewing the 2-up/3-up/4-up on mobile.  If you look at the three screenshots below, the 2-up text will get jumbled between the two icon groups when viewed on mobile, 3-up gets cut off slightly on the right side edge of the display, and 4-up only shows 3 of the icon groups instead of the 4th.

Screen Shot 2022-09-07 at 1 43 40 PM
Screen Shot 2022-09-07 at 1 43 31 PM
Screen Shot 2022-09-07 at 1 42 44 PM

**Comment by Sartxi** [comment]
@amauch-adobe great catch! I have pushed a patch to address the layouts in other viewport sizes. Thank you for noticing this.

**Review by amauch-adobe — APPROVED** [review]
Approved, revalidated mobile/tablet and everything looks good.

**Review by auniverseaway — APPROVED** [review]

**Pull request merged**